default/scripting/specials/planet/monster_nest/monster_nest.macros

```diff
@@ -31,8 +32,11 @@ MONSTER_NEST
             activation = And [
                 Planet
                 Turn low = 10
-                Unowned
-                Random probability = 0.12 * GalaxyMonsterFrequency / 2.0
+                Or [
+                    Unowned
+                    ((GameRule name = "RULE_NESTS_ALWAYS_SPAWN_WILD") > 0)
```

Comment:
switch the lines (rule is independent of planet) for faster processing.
actually moving the Turn condition and the game rule above the Planet should be even faster - @geoffthemedio or do we actually use a cached set of Planets without copying so it is a No-Op ?

Reply:
See comment above... It evaluates a special's activation condition on all objects that have the special. Otherwise not clear what you're asking re: caching.

Reply:
I could move this, but I'm not sure what happens to the Unowned check. At a guess it checks for all unowned objects, instead of all unowned planets? And then looks for planets in unowned objects? Sounds like any improvement in speed is slow, and the logic becomes more confusing (at least to me).

Reply:
As written now, the `Unowned` condition is applied to objects that were put into the initial candidates set for the special in which this `MONSTER_NEST` is used, after all the above conditions have been applied and passed. Those objects must also match either `Unowned` or the `GameRule` test must pass for those objects to pass the overall activation condition. Right now, if there are many such objects, if any of them are not unowned (ie. are owned by an empire), then it will also do the`GameRule` check. You could swap the order and do the `GameRule` check first, and then not need to bother doing the ownership checks on any if the `GameRule` check passes. That said, if there's usually just one or a small number of objects at this point, then possibly checking if they are owned might be faster than doing the `GameRule` check once. Would need to profile to know, and probably not a significant difference anyway.

Not sure what you mean by "it checks for all unowned objects". The condition later in the `And` block is only evaluated on objects that already passed earlier conditions in the same `And` block. "then looks for planets in unowned objects" doesn't make any sense to me...

Reply:
for reference, one can script it to also check if there is an unowned object in the universe, independent of the objects being matched by the `And` condition, but that would require a `Number` or `Statistic If` type condition that match a subcondition independently.

Reply:
Thanks. The main take on this seems to be switching the line: may be faster, may be slower and is unlikely to matter, so I'll keep it as it is?